phoenix-core/src/main/java/org/apache/phoenix/expression/LikeExpression.java

```diff
@@ -289,8 +288,11 @@ public boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr) {
             return true;
         }
 
-        String value = (String) PVarchar.INSTANCE.toObject(ptr, getStrExpression().getSortOrder());
-        boolean matched = pattern.matcher(value).matches();
+        String value = null;
+        if (logger.isDebugEnabled()) {
+            value = (String) PVarchar.INSTANCE.toObject(ptr, getStrExpression().getSortOrder());
```

Comment:
1. The original code has debugging lines, so I try to remains them. We just turn ptr into String when debugs.
2. I don't quite understand what "rename getStrExpression()" means here?